bluemira/base/logs.py

```diff
@@ -69,6 +71,167 @@ def format(self, record) -> str:
+            msg if clean else colourise(msg, color="blue", flush=flush, fmt=fmt),
+            *args,
+            **kwargs,
+        )
+
+    def warning(self, msg, *args, flush: bool = False, fmt: bool = True, **kwargs):
+        """Warning"""
+        return super().warning(
+            colourise(f"WARNING: {msg}", color="orange", flush=flush, fmt=fmt),
+            *args,
+            **kwargs,
+        )
+
+    def error(self, msg, *args, flush: bool = False, fmt: bool = True, **kwargs):
+        """Error"""
+        return super().error(
+            colourise(f"ERROR: {msg}", color="red", flush=flush, fmt=fmt),
+            *args,
+            **kwargs,
+        )
+
+    def critical(self, msg, *args, flush: bool = False, fmt: bool = True, **kwargs):
+        """Critical"""
+        return super().critical(
+            colourise(f"CRITICAL: {msg}", color="darkred", flush=flush, fmt=fmt),
+            *args,
+            **kwargs,
+        )
+
+
+def _bm_print(string: str, width: int = 73) -> str:
+    """
+    Create the text string for boxed text to print to the console.
+
+    Parameters
+    ----------
+    string:
+        The string of text to colour and box
+    width:
+        The width of the box, default = 73 (leave this alone for best results)
+
+    Returns
+    -------
+    :
+        The text string of the boxed text
+    """
+    strings = [
+        " " if s == "\n" and i != 0 else s[:-1] if s.endswith("\n") else s
+        for i, s in enumerate(string.splitlines(keepends=True))
+    ]
+    bw = width - 4
+    t = [
+        wrap(s, width=bw, replace_whitespace=False, drop_whitespace=False)
+        for s in strings
+    ]
+
+    s = [dedent(item) for sublist in t for item in sublist]
+    lines = ["".join(["| "] + [i] + [" "] * (width - 2 - len(i)) + [" |"]) for i in s]
+    h = "".join(["+", "-" * width, "+"])
+    return h + "\n" + "\n".join(lines) + "\n" + h
+
+
+def _bm_print_singleflush(string: str, width: int = 73) -> str:
+    """
+    Create the text string for coloured, boxed text to flush print to the
+    console.
+
+    Parameters
+    ----------
+    string:
+        The string of text to colour and box
+    width:
+        The width of the box, default = 73 (leave this alone for best results)
+
+    Returns
+    -------
+    :
+        The text string of the boxed coloured text to flush print
+    """
+    a = width - len(string) - 2
+    return "| " + string + a * " " + " |"
+
+
+def colourise(
+    string: str,
+    width: int = 73,
+    color: str = "blue",
+    *,
+    flush: bool = False,
+    fmt: bool = True,
+) -> str:
+    """
+    Print coloured, boxed text to the console. Default template for bluemira
+    information.
+
+    Parameters
+    ----------
+    string:
+        The string of text to colour and box
+    width:
+        The width of the box, default = 73 (leave this alone for best results)
+    color:
+        The color to print the text in from `bluemira.base.constants.ANSI_COLOR`
```

Comment:
no fmt, flush doc

Reply:
This is quite a nit, but fmt is so general in this case and when reading this code, I got confused trying to understand what it did and why sometimes fmt=True, and sometimes fmt=False.

I think if this option was changed to bluemira_fmt or bm_fmt, it would be a lot clearer. It's printing in the Bluemira format, the boxed decorated format.

Reply:
I'm trying to move most of this to lower level, I agree generally and I dont think its there yet but a user should never touch this